curvlinops/kfac.py

```diff
@@ -1147,9 +1147,8 @@ def load_state_dict(self, state_dict: Dict[str, Any]):
             input_covariances_keys = set(self._input_covariances.keys())
             gradient_covariances_keys = set(self._gradient_covariances.keys())
             mapping_keys = set(self._mapping.keys())
-            if (
-                input_covariances_keys != mapping_keys
-                or gradient_covariances_keys != mapping_keys
+            if (input_covariances_keys and input_covariances_keys != mapping_keys) or (
+                gradient_covariances_keys and gradient_covariances_keys != mapping_keys
```

Comment:
Could you explain what this does? I am not familiar with this notation for sets.

Reply:
I tried understanding what is happening with the following:
```python
a = {1,2}
b = {2,3}

a and b # returns {2,3}
```

Reply:
In Python `!=` has higher precedence than `and`, i.e. each of the expressions in the parentheses here are of type `set and bool`. A set evaluated as a boolean will return `True` if it is non-empty and `False` if it is empty. Hence, this expression is first checking if the set is non-empty and then if it matches the `mapping_keys`. Does this make sense or am I missing something?